kivy/core/text/__init__.py

```diff
@@ -263,11 +272,34 @@ def __init__(
         kwargs_get = kwargs.get
         options['color'] = color or (1, 1, 1, 1)
         options['outline_color'] = outline_color or (0, 0, 0, 1)
-        options['padding'] = kwargs_get('padding', (0, 0))
-        if not isinstance(options['padding'], (list, tuple)):
-            options['padding'] = (options['padding'], options['padding'])
-        options['padding_x'] = kwargs_get('padding_x', options['padding'][0])
-        options['padding_y'] = kwargs_get('padding_y', options['padding'][1])
+
+        options['padding'] = kwargs_get('padding', [0, 0, 0, 0])
+        if isinstance(options['padding'], (int, float)):
+            options['padding'] = [options['padding']] * 4
+        elif (
+            isinstance(options['padding'], (list, tuple))
+            and len(options['padding']) != 4
+        ):
+            if len(options['padding']) == 1:
+                options['padding'] = options['padding'] * 4
```

Comment:
Previously `len(options['padding']) == 1` was not supported.

Do we really want to introduce this chance?